avm/res/network/p2s-vpn-gateway/main.bicep

```
@@ -0,0 +1,192 @@
+metadata name = 'P2S VPN Gateway'
+metadata description = 'This module deploys a Virtual Hub P2S Gateway.'
+metadata owner = 'Azure/module-maintainers'
+
+@description('Required. The name of the P2S VPN Gateway.')
+param name string
+
+@description('Required. Location where all resources will be created.')
+param location string = resourceGroup().location
+
+@allowed([
+  'noneRouteTable'
+  'defaultRouteTable'
+])
+@description('Conditional. The name of the associated route table. Required if deploying in a Secure Virtual Hub; cannot be a custom route table.')
+param associatedRouteTableName string?
+
+@description('Optional. The names of the route tables to propagate to the P2S VPN Gateway.')
+param propagatedRouteTableNames array = []
+
+@description('Optional. The custom DNS servers for the P2S VPN Gateway.')
+param customDnsServers array = []
+
+@description('Optional. The routing preference for the P2S VPN Gateway, Internet or Microsoft network.')
+param isRoutingPreferenceInternet bool?
+
+@description('Required. The name of the P2S Connection Configuration.')
+param p2SConnectionConfigurationsName string?
+
+@description('Optional. Enable/Disable Internet Security; "Propagate Default Route".')
+param enableInternetSecurity bool?
+
+@description('Optional. The Resource ID of the inbound route map.')
+param inboundRouteMapResourceId string?
+
+@description('Optional. The Resource ID of the outbound route map.')
+param outboundRouteMapResourceId string?
+
+@description('Optional. The Labels to propagate routes to.')
+param propagatedLabelNames array = []
+
+@description('Required. The address prefixes for the VPN Client Address Pool.')
+param vpnClientAddressPoolAddressPrefixes array = []
+
+@description('Required. The resource ID of the gateways virtual hub.')
+param virtualHubId string
```

```suggestion
param virtualHubResourceId string
```
As per [ref](https://azure.github.io/Azure-Verified-Modules/specs/shared/#id-snfr22---category-inputs---parametersvariables-for-resource-ids)

---

Corrected all instances of that parameter name